File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.java
class Solution {
    public int maximumSum(int[] nums) {
        HashMap<Integer,Integer> map=new HashMap<>();
        int maxi=-1;
        for(int i=0;i<nums.length;i++){
           int sum_digits=f(nums[i]);
           if(map.containsKey(sum_digits)){
             int ans=map.get(sum_digits);
             if(nums[i]>map.get(sum_digits)){
                map.put(sum_digits,nums[i]);
             }
             ans+=nums[i];
             maxi=Math.max(maxi,ans);
           }
           else{
            map.put(sum_digits,nums[i]);
           }
        }
        return maxi;
    }
    public static int f(int n){
        int ans=0;
        while(n>0){
            int rem=n%10;
            ans+=rem;
            n=n/10;
        }
        return ans;
    }
}

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums: List[int]) -> int:
        new_arr=[str(i) for i in nums]
        print(new_arr)
        arr=[]
        for i in range(len(nums)):
            s=0
            for j in range(len(new_arr[i])):
                s+=int(new_arr[i][j])

            arr.append(s)

        #[9,7,9,4,7]

        print(arr)
        d=dict()
        maxi_sum=-1
        for i in range(len(arr)):
            if arr[i] not in d:
                d[arr[i]]=i

            else:
                
                prev_ind=d.get(arr[i])
                curr_sum=nums[i]+nums[prev_ind]
                maxi_sum=max(curr_sum,maxi_sum)
                if nums[prev_ind]<nums[i]:
                    d[arr[i]]=i

        return maxi_sum

            

            

            


        

    

